Sync/StorageClient.swift

```diff
@@ -605,6 +616,210 @@ public class Sync15StorageClient {
     }
 }
 
+public class Sync15BatchClient<T: CleartextPayloadJSON> {
+    private let config: InfoConfiguration
+
+    private var records: [Record<T>] = []
+    private let collectionClient: Sync15CollectionClient<T>
+    private let collectionURI: NSURL
+
+    init(config: InfoConfiguration, collectionURI: NSURL, collectionClient: Sync15CollectionClient<T>) {
+        self.config = config
+        self.collectionURI = collectionURI
+        self.collectionClient = collectionClient
+    }
+
+    public func addRecords(records: [Record<T>]) {
+        log.debug("Adding \(records.count) records into batch")
+        self.records += records
+    }
+
+    // TODO: use I-U-S.
+    // Each time we do the storage operation, we might receive a backoff notification.
+    // For a success response, this will be on the subsequent request, which means we don't
+    // have to worry about handling successes and failures mixed with backoffs here.
+
+    public func commit(ifUnmodifiedSince: Timestamp? = nil, onCollectionUploaded: (POSTResult -> Void)) -> Success {
```

Comment:
I think IUS should be part of the initializer, no? It's known when the batch begins, and the caller can only call `commit` once.

If multiple entire batches need to be sent (> 10,000 records) then the timestamp will be chained through.